r2ai/interpreter.py

```diff
@@ -756,6 +760,71 @@ def compress_messages(self, messages):
     # if msglen > 4096:
     #   ¡print("Query is too large.. you should consider triming old messages")
     return messages
+  
+  def process_tool_calls(self, tool_calls):
+    self.messages.append({ "tool_calls": tool_calls, "role": "assistant" })
+    for tool_call in tool_calls:
+      res = ''
+      if tool_call["function"]["name"] == "r2cmd":
+        args = json.loads(tool_call["function"]["arguments"])
+        sys.stdout.write('\x1b[1;32mRunning \x1b[4m' + args["command"] + '\x1b[0m\n')
+        # r2lang.cmd('e scr.color=0')
+
+        res = r2lang.cmd(args["command"])
+        sysprint(res)
+      elif tool_call["function"]["name"] == "run_python":
+        args = json.loads(tool_call["function"]["arguments"])
+        # save to file
+        with open('temp.py', 'w') as f:
+          f.write(args["command"])
+        sys.stdout.write('\x1b[1;32mRunning \x1b[4m' + "#!python temp.py" + '\x1b[0m\n')
+        sysprint(args["command"])
+        sysprint("")
+        r2lang.cmd('#!python temp.py > temp_output')
```

Comment:
you can also redirect the output to an internal file like this:

```
r2lang.cmd('#!python temp.py > $tmp')
output = r2lang.cmd('cat $tmp')
```

the files starting with $ are internal to r2, so wont create any file in disk.. 

if reading from disk is necessary better use the slurp api from utils.py, and dont forget to remove it after using it

Reply:
oh nice, thanks!